pagopa/io-app
# feat(Services): [#176676680,#176606850,#176513922] Improve services caching and loading

## Short description
This PR improves services loading and caching

### service caching
- services detail are persisted with **no cache expiration**. A service will be deleted only when it is no longer visible (the app has a service that is not included in the API response `api/v1/services`)

### service loading
- in certain scenario a lot of service detail load action are dispatched, i.e when all visibile services are loaded. To avoid this burst of actions (due to this also mixpanel tracking is overloaded) a dedicated **loadServicesDetail** has been created
- all services detail request are handled by using a configured amount of workers that consume these event from a dedicated channel

### how to test
- **caching** 
1. login
2. load messages / services
3. logout _or_ session expired
4. login
5. services detail don't be loaded again

- **detail loading** 
1. setup io-dev-server to [serve few hundreds](https://github.com/pagopa/io-dev-api-server/blob/c67fb08e9bbff77ac1de8a755adb353387cce142/src/routers/service.ts#L12) of service
2. reload the app
3. the app should be smooth




https://user-images.githubusercontent.com/822471/106465794-bf2bec00-649a-11eb-9964-f41c2f534591.mp4
